### ISSUE TITLE
Issue with ingest

### ISSUE THREAD

**Issue by plrlhb12** [scverse/scanpy#1181]
I am trying to integrate data using ingest. I followed exactly the same steps and scripts as described in the scanpy tutorial. However, I got the error message that ‘UMAP’ object has no attribute ‘_input_distance_func’ every time when I ran the command of sc.tl.ingest(adata, adata_ref, obs=‘louvain’). I have the same problem even with using the example pbmc3k_processed and pbmc3k_processed as in the tutorial.

```
adata_ref = sc.datasets.pbmc3k_processed()
adata = sc.datasets.pbmc68k_reduced()
var_names = adata_ref.var_names.intersection(adata.var_names)
adata_ref = adata_ref[:, var_names]
adata = adata[:, var_names]
sc.pp.pca(adata_ref)
sc.pp.neighbors(adata_ref)
sc.tl.umap(adata_ref)

# problem occurs here
sc.tl.ingest(adata, adata_ref, obs='louvain')
...
```

```
AttributeError                            Traceback (most recent call last)
<ipython-input-12-27e22cc8f823> in <module>()
----> 1 sc.tl.ingest(adata, adata_ref, obs='louvain')

3 frames
/usr/local/lib/python3.6/dist-packages/umap/umap_.py in transform(self, X)
   2006             try:
   2007                 # sklearn pairwise_distances fails for callable metric on sparse data
-> 2008                 _m = self.metric if self._sparse_data else self._input_distance_func
   2009                 dmat = pairwise_distances(
   2010                     X, self._raw_data, metric=_m, **self._metric_kwds

AttributeError: 'UMAP' object has no attribute '_input_distance_func'
```

#### Versions:

> scanpy==1.4.6 anndata==0.7.1 umap==0.4.1 numpy==1.18.2 scipy==1.4.1 pandas==1.0.3 scikit-learn==0.22.2.post1 statsmodels==0.10.2 python-igraph==0.8.0 louvain==0.6.1

**Comment by Koncopd** [comment]
Hi, @plrlhb12 .
Yes, this is a known problem. The easiest fix for now is to use umap 0.3.9 instead of 0.4.1.
You can also install scanpy from github where it is fixed or just wait for a new scanpy release.

**Comment by plrlhb12** [comment]
Hi Sergei,

Thank you very much for your fast reply. Do you mean I can still use the latest version of scanpy but installing a lower version of umap?

I tried different versions of scanpy, including pastiest, stable, 1.4.5, 1.4.5post3, 1.4.4post1... They seem to either have different error messages or packages not compatible.   Do you know which version of the scanpy has it fixed?

Thank you for your kind help.

Best regards,

Lirong

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Sergei R. <notifications@github.com>
发送时间: Wednesday, April 22, 2020 12:44:36 PM
收件人: theislab/scanpy <scanpy@noreply.github.com>
抄送: plrlhb12 <lrpeng@hotmail.com>; Mention <mention@noreply.github.com>
主题: Re: [theislab/scanpy] Issue with ingest (#1181)


Hi, @plrlhb12<https://github.com/plrlhb12> .
Yes, this is a known problem. The easiest fix for now is to use umap 0.3.9 instead of 0.4.1.
You can also install scanpy from github where it is fixed or just wait for a new scanpy release.

―

**Comment by plrlhb12** [comment]
It works now by changing umap to 0.3.9. Thanks a lot!

Best regards,

Lirong

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Sergei R. <notifications@github.com>
发送时间: Wednesday, April 22, 2020 12:44:36 PM
收件人: theislab/scanpy <scanpy@noreply.github.com>
抄送: plrlhb12 <lrpeng@hotmail.com>; Mention <mention@noreply.github.com>
主题: Re: [theislab/scanpy] Issue with ingest (#1181)


Hi, @plrlhb12<https://github.com/plrlhb12> .
Yes, this is a known problem. The easiest fix for now is to use umap 0.3.9 instead of 0.4.1.
You can also install scanpy from github where it is fixed or just wait for a new scanpy release.

―
